pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -220,16 +252,116 @@ func (h *handler) OnRemove(_ string, obj runtime.Object) (runtime.Object, error)
 		return nil, err
 	}
 
+	// In the event we are removing an etcd node (as indicated by the etcd-role label on the node), we must safely remove the etcd node from the cluster before allowing machine deprovisioning
+	if val, ok := d.Map("metadata", "labels")["rke.cattle.io/etcd-role"]; ok {
+		if val.(string) == "true" {
+			// we need to block removal until our the v1 node that corresponds has been removed
+			clusterName := d.Map("metadata", "labels")[CapiMachineLabel].(string)
```

Comment:
To avoid a potential panic.
```suggestion
			clusterName, _ := d.Map("metadata", "labels")[CapiMachineLabel].(string)
```